unison-runtime/src/Unison/Runtime/Machine.hs

```diff
@@ -547,6 +549,15 @@ exec !env !denv !_activeThreads !stk !k _ (BPrim2 TRCE i j)
 exec !_ !denv !_trackThreads !stk !k _ (BPrim2 op i j) = do
   stk <- bprim2 stk op i j
   pure (denv, stk, k)
+exec !_ !denv !_activeThreads !stk !k _ (RefCAS refI ticketI valI) = do
+  (ref :: IORef Val) <- peekOffBi stk refI
+  (ticket :: Atomic.Ticket Val) <- peekOffBi stk ticketI
+  v <- peekOff stk valI
```

Comment:
Need to evaluate `v`, not `ticket`.

Reply:
Note: we looked into this and surprisingly this evaluate is required.